src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/UnsupportedStatement.scala

```diff
@@ -0,0 +1,49 @@
+package net.snowflake.spark.snowflake.pushdowns.querygeneration
+
+import net.snowflake.spark.snowflake._
+import org.apache.spark.sql.catalyst.expressions.{
+  Attribute,
+  Expression,
+  PythonUDF,
+  ScalaUDF
+}
+import org.apache.spark.sql.execution.aggregate.ScalaUDAF
+
+/**
+  * This class is used to catch unsupported statement and raise an exception
+  * to stop the push-down to snowflake.
+  */
+private[querygeneration] object UnsupportedStatement {
+
+  final val EXCEPTION_MESSAGE = "Meet unsupported statement"
```

Comment:
Should this actually be an exception? It should be enough to simply send a telemetry message. Exceptions can be big and scary, and we don't want to alarm users if we simply failed to translate something.

Reply:
Ok nevermind, I just saw your catch clause up there. This should be fine.

Reply:
And also nit: change the message to 'pushdown failed'

Reply:
1. It's necessary to raise an exception here and let outer function to catch the exception and send the telemetry message. The reason is that we only need to send telemetry message if there are any snowflake tables. We don't know whether there are any snowflake table here.
2. I will change the message as 'pushdown failed'.